frontend/src/metabase/query_builder/components/notebook/NotebookStep.jsx

```diff
@@ -147,7 +152,12 @@ export default class NotebookStep extends React.Component {
 
     return (
       <ExpandingContent isInitiallyOpen={!isLastOpened} isOpen>
-        <Box mb={[1, 2]} pb={[1, 2]} className="hover-parent hover--visibility">
+        <Box
+          mb={[1, 2]}
+          pb={[1, 2]}
+          className="hover-parent hover--visibility"
+          data-testid={getTestId(step)}
+        >
```

Comment:
@nemanjaglumac this change adds a test ID to every notebook step. It looks pretty difficult to select a specific step in Cypress, so I tried to simplify this just a bit. Please share your feedback.

Test ID structure:

1. Just a "step" word
2. Step's type (data / join / filter / summarize, etc.)
3. Step's stage index. If a question has a complete Summarize section, you can add joins, filters, sorting, and other breakouts on top of that query. In this case, the notebook is split into "stages". Step's stage index identifying to which stage the step belongs
4. Step's item index. If a question has let's say a few joins in a row, they would have unique item indexes

See a screenshot with an example:

![CleanShot 2021-09-01 at 13 19 55@2x](https://user-images.githubusercontent.com/17258145/131655489-9741fd25-192e-4137-ac2a-472a27e07b3a.png)

Reply:
Thinking out loud: maybe we can also have a cypress helper looking something like that:

```js
function findNotebookSection(type, { stage = 0, index = 0 }) {
   return cy.findByTestId(`step-${type}-${stage}-${index}`);
}

// Usage

findNotebookSection("join") // returns first join section
findNotebookSection("join", { index: 1 }) // returns second join section
findNotebookSection("join, { stage: 1 }) // returns first join section in the second stage
```

Reply:
Thank you for putting this together. It is very much needed.
Just a qq: shouldn't `step-filter` from the screenshot be `0-0`?

This seems good to me as the first step, and we can work on the Cypress helper next.
With just slightly better HTML and CSS structure, we might not even need data-testids, but that's a whole different story.

Once again, let's go with this.
Thank you <3

Reply:
Oh my bad, it would be `step-filter-0-0`. Happy to help!